src/libasr/codegen/wasm_to_x86.cpp

```diff
@@ -329,9 +348,92 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
     void visit_I32LeS() { handle_I32Compare<&X86Assembler::asm_jle_label>(); }
     void visit_I32Ne() { handle_I32Compare<&X86Assembler::asm_jne_label>(); }
 
-    void visit_F64Const(double Z) {
-        float z = Z; // down cast 64-bit double to 32-bit float
-        std::string label = "float_" + std::to_string(z);
+    void visit_I64Const(int64_t value) {
+        m_a.asm_push_imm32(value);
+    }
+
+    void visit_I64ExtendI32S() {
+        // empty, since all i32's are already considered as i64's currently.
+    }
```

Comment:
For `i64` types, we are currently storing and managing them as `i32`. This is way we simply get the backend working with `i64`'s and passing the `CI`. Please share if this approach is fine.

Reply:
Yes, that's perfect.